python-package/lightgbm/compat.py

```diff
@@ -316,6 +315,26 @@ class pa_Table:  # type: ignore
         def __init__(self, *args: Any, **kwargs: Any):
             pass
 
+    class pa_compute:  # type: ignore
+        """Dummy class for pyarrow.compute."""
+
+        all = None
+        equal = None
+
+    pa_chunked_array = None
+    arrow_is_boolean = None
+    arrow_is_integer = None
+    arrow_is_floating = None
+
+
+"""cffi"""
+try:
+    from pyarrow.cffi import ffi as arrow_cffi
+
+    CFFI_INSTALLED = True
+except ImportError:
+    CFFI_INSTALLED = False
+
     class arrow_cffi:  # type: ignore
         """Dummy class for pyarrow.cffi.ffi."""
```

Comment:
Why do we need

```
        CData = None
        addressof = None
        cast = None
        new = None
```
class members?

Reply:
`CData` is needed for type hinting:

https://github.com/microsoft/LightGBM/blob/9f1af051b44564eaab2bebe1612c6a52217bb32b/python-package/lightgbm/basic.py#L416

But I think the others could be safely removed. It's only showing up in the diff in this PR because this code is being moved around... so this was missed in earlier PRs (I guess #6034).

Reply:
Removed all but `CData` in https://github.com/microsoft/LightGBM/pull/6785/commits/7396613590c06e03876612df30f09e9e0ee2e44a